File: software/examples/python/example_simple.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_line import Line

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    line = Line(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get current reflectivity
    reflectivity = line.get_reflectivity()

    print('Reflectivity: ' + str(reflectivity))

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleSimple.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletLine = require('Tinkerforge/BrickletLine');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'j6q';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var line = new BrickletLine(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        //Get current reflectivity
        line.getReflectivity(
            function(reflectivity) {
                console.log('Reflectivity: '+reflectivity);
            },
            function(error) {
                console.log('Error: '+error);
            }
        );
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

